interfaces/builtin/basedeclaration.go

```diff
@@ -267,6 +267,10 @@ slots:
       slot-snap-type:
         - core
     deny-auto-connection: true
+  lxd-client:
+    allow-installation:
+      slot-snap-type:
+        - core
```

Comment:
After changing to a slot implementation, change this to:

```
lxd:
  allow-installation: false
  deny-connection: true
  deny-auto-connection: true  
```

This will require the slot side implementation to need a snap declaration to even be installed and requires a plugging snap to also need a snap declaration to connect to lxd. I fully agree with @stgraber's comments regarding who privileged this interface is, and adjusting the base declaration to do this addresses those concerns. You'll of course need to adjust the testsuite changes accordingly.